core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/AddClusterRolesDecorator.java

```diff
@@ -140,22 +156,35 @@ private static Set<PolicyRule> getClusterRolePolicyRulesFromDependentResources(Q
                             resourceGroup = gvk.getGroup();
                             resourcePlural = gvk.getPluralOrDefault();
                         }
-
-                        // if we use SSA and the dependent resource class is not excluded from using SSA, we also need PATCH permissions for finalizer when the dependent resource is creatable
-                        if (isCreator && cri.getConfigurationService().shouldUseSSA(kubeResource)) {
-                            verbs.add(RBACVerbs.PATCH);
-                        }
                     } catch (Exception e) {
-                        log.warn("Ignoring " + dependentResourceClass.getName()
-                                + " for generic resource role processing as it cannot be instantiated", e);
+                        ignore = true;
+                        log.warn(" Ignoring dependent " + dependentResourceClass.getName()
+                                + "because it couldn't be instantiated as it doesn't provide a no-arg constructor, preventing its group and plural to be determined.");
```

Comment:
```suggestion
                                + " because it couldn't be instantiated as it doesn't provide a no-arg constructor, preventing its group and plural to be determined.");
```